### ISSUE TITLE
Sim enemies unusually difficult at higher framerates (Tick Rate Divisor of 0)? / Sim enemies easier at Lower Framerates (TRD of 2)?

### ISSUE THREAD

**Issue by togera13** [fgsfdsfgs/perfect_dark#263]
I am mainly talking about the multiplayer's sims & challenges here, I am _VERY_ confident that higher framerates make them react _FAR_ faster.

I am not sure if I am just tweaking from getting my butt kicked repeatedly on challenge 16 but changing the game to TickRateDivisor of 2 I think it was, where the framerate is set to 30 I was able to beat the challenge far easier, while at a TickRateDivisor of 0 or 144 fps I would get flat out destroyed the second a pixel of the enemy was visible it felt like.

I don't know if this is a me thing or if it is actually easier the lower the framerate, I had the same issue with the challenge where you protect the ai with one shots on, as it seemed like the enemy ai had a far more reasonable reaction time & aim at 30 fps TRD 2; while being nigh unstoppable feeling at 144fps TRD 0.

Though the game does seem to be just naturally hard, but I don't know if this is actually a problem or not, but trying to play it with my Dad he kinda had the same issue it felt like where it felt like the enemies just responded too fast.

Again I could just be bad, but I am pretty confident that dropping to 30FPS & TRD of 2 makes the game easier...

& if this is somehow intentional would we be able to get a difficulty modifier that lets us manually change to the slightly easier difficulty for higher framerates? My family is not full of professional gamers, but I do like high framerates...

**Comment by zerocoregithub** [comment]
haven't noticed a difficulty spike in SP, but Perfect Sims in MP are next time impossible to react to at high frame rates. Their reaction time, aim and damage output is insane. Feels far more difficult than anything I remember from my N64 days.

**Comment by fgsfdsfgs** [comment]
I'm assuming this is because sims' reaction time is probably measured in frames or something. I'll take a look.

**Comment by JAMoreno-Larios** [comment]
Hi all,

I checked file perfect_dark/src/game/bot.c, lines 45 to 62, I found that the difficulty is indeed tied to TICKS()

I'm completely new to the codebase, I'm assuming TICKS means frames in this context, is that correct?

**Comment by fgsfdsfgs** [comment]
Not exactly. `TICKS()` is a macro that's used to scale tick counts between PAL and NTSC builds, but time can be measured in either 60hz ticks or 240hz ticks, and those can be calculated in different ways, some framerate independent.
The problem is at least partially in that code though. For the last two difficulties some of the reaction times are set to 0, which means sims won't wait for any amount of ticks in those cases, which means they'll react the very next frame. And at high framerates that's very fast.

**Comment by JamesPConnolly91** [comment]
I can attest to this. I was playing challenges on my Steam Deck OLED, and they were all push overs until I got to 16, which guides online say includes the first PerfectSim in the challenges so far. The Sim blasts you the millisecond it sees you, it almost appears preemptive at times. I don't remember the challenge being this difficult on the original. 

I see the creator has already responded, but just wanted to put my hand up in attendance, if you will. Thank you. 🙂

**Comment by GitMoDu** [comment]
On the latest version, I was doing the Combat Simulator challenges, and some of the sims are nearly impossible. Then I remembered I'm playing on 120Hz and that might not help.

**Comment by FrostKnight** [comment]
> On the latest version, I was doing the Combat Simulator challenges, and some of the sims are nearly impossible. Then I remembered I'm playing on 120Hz and that might not help.

 I have beaten every challenge going all out on them before, but not at 120 Hz to my knowledge. That sounds really rough.

On an unrelated note, I hesitate to add new threads given how many I have made today, but I wonder how many simulants this game can handle without crashing on pc. lol.
